### PR TITLE
Correctly generate missing tokens caught during Panda compilation

### PR DESCRIPTION
**Pull request by coopbri**
##### Task link: https://linear.app/omnidev/issue/OMNI-54/fix-missing-tokens-issue

- Fixed missing tokens error by correctly generating missing semantic token paths. Certain token accessors, e.g. `brand.primary`, didn't resolve correctly because of the prefix, where the semantic color token generator function was always fixed to top level. I added an optional `prefix` string to handle these cases
- Added missing Panda studio dependency
- Set Panda validation failures to error instead of warn

## Test Steps

- Verify missing tokens warning is not present (`bun panda check` works for quick feedback loop)
- Verify `brand` and `omni` color tokens work. E.g. `brand.primary.500`. This can be tested in a Storybook story, for example the `Square.stories.tsx` file changing background color. The tokens can also be seen in Panda studio which can be ran with `bun studio`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
   },
   "exports": {
     ".": {
+      "types": "./build/sigil.d.ts",
```
Comment (coopbri) [inline package.json:19]: I received a validation warning about this -- apparently, `types` needs to precede `require`/`import`

Comment (hobbescodes) [reply package.json:19]: Nice! Yeah I have seen some discussions on this before (will try to find some resources), but didn't even think about it!

Comment (coopbri) [reply package.json:19]: I had no idea. Some tool caught it for us after I upgraded deps, guessing tsup/esbuild

File: src/lib/theme/extensions/semanticTokens/colors.semantic.ts [hunk 2]
```diff
@@ -15,33 +15,52 @@ const NEUTRAL_PALETTE = "silver";
...
+   * @param prefix - The optional prefix to prepend to the token path.
+   * @returns The token path for the given color.
+   */
+  const calculateTokenPath = ({
+    color,
+    prefix,
+  }: {
+    color: string;
+    prefix?: string;
+  }) => (prefix ? `{colors.${prefix}.${color}}` : `{colors.${color}}`);
```
Comment (hobbescodes) [inline src/lib/theme/extensions/semanticTokens/colors.semantic.ts:37]: Nice!

File: package.json [hunk 3]
```diff
@@ -57,27 +57,28 @@
     "@commitlint/config-conventional": "^19.5.0",
     "@commitlint/types": "^19.5.0",
     "@omnidev/knit": "^0.1.2",
-    "@pandacss/dev": "0.46.1",
-    "@pandacss/types": "0.46.1",
-    "@storybook/addon-a11y": "^8.3.5",
+    "@pandacss/dev": "0.47.0",
+    "@pandacss/studio": "^0.47.0",
```
Comment (coopbri) [inline package.json:61]: @hobbescodes FYI Panda requires this for studio now. Nice having the split packages

Comment (coopbri) [reply package.json:61]: It (@pandacss/dev) also warns you if you try to run studio without it so no need to "keep track" of this info

Comment (hobbescodes) [reply package.json:61]: Nice! Ty, didnt know.

**Review by hobbescodes — APPROVED** [review]
Very nice!

**Pull request merged**